graphql_compiler/tests/test_compiler.py

```diff
@@ -1308,6 +1308,120 @@ def test_in_collection_op_filter_with_optional_tag(self):
 
         check_test_data(self, test_data, expected_match, expected_gremlin)
 
+    def test_intersects_op_filter_with_variable(self):
+        test_data = test_input_data.intersects_op_filter_with_variable()
+
+        expected_match = '''
+            SELECT
+                Animal___1.name AS `animal_name`
+            FROM (
+                MATCH {{
+                    class: Animal,
+                    where: ((intersect(alias, {wanted}).asList().size() > 0)),
+                    as: Animal___1
+                }}
+                RETURN $matches
+            )
+        '''
+        expected_gremlin = '''
+            g.V('@class', 'Animal')
+            .filter{it, m -> it.alias.intersects($wanted)}
```

Comment:
Two issues here:
- The Gremlin method is `intersect` not `intersects`.
- `intersect` returns the intersection itself, which is not a boolean and produces an error on evaluation:
`java.lang.ClassCastException: java.util.LinkedHashSet cannot be cast to java.lang.Boolean`

The easiest compilation option here probably is:
```
.filter{it, m -> (!it.alias.intersects($wanted).empty)}
```

That means the Gremlin format for `intersects` would be: 
```
u'(!{left}.{operator}({right}).empty)'
```